### PR TITLE
Download GWSW HydX datasets through GWSW API

### PR DESCRIPTION
**Pull request by leendertvanwolfswinkel**

### CODE REVIEW

File: processing/download_hydx.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    :param dataset_name:
+    :param target_directory:
+    :param wait_times:
+    :param feedback:
+    :return:
+    """
+    wait_times = wait_times or [1]
+    feedback = feedback or MockQgsProcessingFeedback()
+    request_hydx_url = f"https://apps.gwsw.nl/hyd/makehydx?filter={dataset_name}&delim=pk"
+    hydx_headers = {"Authorization": "Bearer requestN&S"}
```
Comment (caspervdw) [inline processing/download_hydx.py]: What is this? "requestN&S" is the authorization header? (is this a security problem?)

For sure constants like these should be defined at the top of the file, you might want to configure that.

**Review by caspervdw — APPROVED** [review]

File: processing/download_hydx.py (hunk 1 above)
Comment (leendertvanwolfswinkel) [reply processing/download_hydx.py]: This is the login key. A security issue I can see here is that other people could access the GWSW API using this key, without our consent. I asked Rioned if they have any other way to organize this, but this is their only solution at the moment. Do you think this security risk is acceptable?

I have now defined this as a global variable (KEY) at the top of the file.

Comment (caspervdw) [reply processing/download_hydx.py]: So we let any user of ThreeDiToolbox authenticate as N&S on app.gwsw.nl ?  Also, this is public, so anyone can now already authenticate as N&S.

Apparently Rioned expects users to authenticate. Maybe there are limits or they just want to keep statistics. The logical solution is that if a user wants data from app.gwsw.nl, he/she needs to get its own authorization key. Is that something that Rioned could support? I expect this to be an input field in the download tool.

Comment (leendertvanwolfswinkel) [reply processing/download_hydx.py]: I think the security risk is for Rioned, not for us. I asked them about this and they suggested this approach. They don't have any infrastructure for users to create there own key atm.

**Pull request merged**